pkg/oonf_api/Makefile

```diff
@@ -8,14 +8,34 @@ MODULE:=$(PKG_NAME)
 # GCC 7.x fails on (intentional) fallthrough, thus disable implicit-fallthrough.
 CFLAGS += -Wno-implicit-fallthrough
 
+COMBINED_ARCHIVE = $(BINDIR)/$(MODULE).a
+
 .PHONY: all
 
 all:
 	"$(MAKE)" -C $(PKG_BUILDDIR)
-	"$(MAKE)" $(BINDIR)/$(MODULE).a
+	"$(MAKE)" $(COMBINED_ARCHIVE)
+
+PARTIAL_ARCHIVES = $(wildcard $(BINDIR)/oonf_*.a)
+
+$(COMBINED_ARCHIVE): $(BINDIR)/$(MODULE).mri $(PARTIAL_ARCHIVES)
+	ar -M < $<
+
+define ADDLIB_TEMPLATE
+addlib $1
```

Comment:
I'm not that versed in `make` syntax, but this seems like it would do nothing.
Would this not just evaluate to
```
$(foreach a,$2,$(call addlib,$a))
```

Reply:
The "foreach" is in line 31 below. The definition is done via "define" instead of "=" because "define" allows me to put a newline at the end.

Reply:
Essentially, this is the same as `ADDLIB_TEMPLATE = addlib $1 \n`, if such a thing (`\n`) were allowed in make, which it isn't.

Reply:
Ah I see, didn't realize the newline was significant here!